File: 0931-minimum-falling-path-sum/0931-minimum-falling-path-sum.java
class Solution {
    public int minFallingPathSum(int[][] matrix) {
        int row = matrix.length;
        int col = matrix[0].length;
        
        for (int i = 1; i < row; i++) {
            for (int j = 0; j < col; j++) {
                int a = Integer.MAX_VALUE, b = Integer.MAX_VALUE;
                if (j - 1 >= 0) {
                    a = matrix[i-1][j-1];
                }
                if (j + 1 < col) {
                    b = matrix[i-1][j+1];
                }
                matrix[i][j] += Math.min(Math.min(matrix[i-1][j], a), b);
            }
        }
        
        int result = Integer.MAX_VALUE;
        
        for (int i = 0; i < col; i++) {
            result = Math.min(result, matrix[row - 1][i]);
        }
        return result;
    }
}

File: LeetCode/0931-minimum-falling-path-sum/0931-minimum-falling-path-sum.py
class Solution:
    def minFallingPathSum(self, matrix: List[List[int]]) -> int:
        row = len(matrix)
        col = len(matrix[0]) if row > 0 else 0

        for i in range(1, row):
            for j in range(col):
                a = float('inf') if j - 1 < 0 else matrix[i-1][j-1]
                b = float('inf') if j + 1 >= col else matrix[i-1][j+1]
                matrix[i][j] += min(matrix[i-1][j], a, b)

        result = float('inf')
        for i in range(col):
            result = min(result, matrix[row-1][i])

        return result